docs/design/declaring_entities.md

```diff
@@ -105,10 +105,11 @@ There are two forms of the `extern` modifier:
 -   On an owning declaration, `extern` limits access to the definition.
     -   The entity must be directly imported in order to use of the definition.
     -   An `extern library` declaration is optional.
--   On a non-owning declaration, `extern library` provides a tool for breaking
-    dependency cycles.
+-   On a non-owning declaration, `extern library` allows references to an entity
+    without depending on the owning library.
     -   The library name indicates where the entity is defined.
-    -
+    -   This may be used to break dependency cycles between libraries, which
+        could be used to improve build parallelization.
```

Comment:
It seems like this might be alternatively be used to improve build scalability by reducing dependencies. This is independent of breaking dependency cycles, and isn't specifically about build parallelization.

```suggestion
    -   This may be used to break dependency cycles between libraries, which
        could be used to improve build parallelization.
    -   This may be used to reduce transitive dependencies, improving build
        scalability.
 ```

Reply:
How about this rephrasing... (to copy here)

```
    -   This can be used to improve build performance, such as by splitting out
        a declaration in order to reduce a library's dependencies.
```

Note, trying to step aside from specific ways it might improve build performance because I want these bullets to focus more on what the feature does, versus how it might be used.

Reply:
I'm fine with the new text, but don't you want something about that it can allow you to break cycles too?

Reply:
To me, that sort of feels like details where if we really want to explain how to use the feature this way, we should have more examples of refactorings to do.

Reply:
i.e., I'm trying to figure out the right line to draw between proposal advocating for adoption of the feature, design trying to explain what it is, and user guide explaining how to apply it in anger.

Reply:
I feel like the design shouldn't just say what it is, but also the why, and the purpose of the feature is definitely in scope.